velox/functions/prestosql/ArraySort.cpp

```diff
@@ -314,10 +314,13 @@ class ArraySortLambdaFunction : public exec::VectorFunction {
     auto sortedElements = BaseVector::wrapInDictionary(
         nullptr, indices, newNumElements, flatArray->elements());
 
+    // Set nulls for rows not present in 'rows'.
+    BufferPtr newNulls = addNullsForUnselectedRows(flatArray, rows);
+
     VectorPtr localResult = std::make_shared<ArrayVector>(
         flatArray->pool(),
         flatArray->type(),
-        flatArray->nulls(),
+        std::move(newNulls),
         flatArray->size(),
```

Comment:
What happens when you try and access a row > rows.end()  in the result ; Assuming size > rows.end() , this might be set with a non null value.

Reply:
NVM - We are setting newNulls with everything > rows.end() as null...